include/language-factory.php

```diff
@@ -260,10 +264,27 @@ private static function get_flag( $flag_code, $name, $slug, $locale ) {
 		 * @param string $slug   The language code.
 		 * @param string $locale The language locale.
 		 */
-		$title = apply_filters( 'pll_flag_title', $name, $slug, $locale );
+		$title  = apply_filters( 'pll_flag_title', $name, $slug, $locale );
+		$return = array();
 
+		/**
+		 * @var array{
+		 *     flag: array{
+		 *         url: string,
+		 *         src: string,
+		 *         width?: positive-int,
+		 *         height?: positive-int
+		 *     },
+		 *     custom_flag?: array{
+		 *         url: non-empty-string,
+		 *         src: non-empty-string,
+		 *         width?: positive-int,
+		 *         height?: positive-int
+		 *     }
+		 * } $flags
+		 */
```

Comment:
A proper PHPstan return type on `PLL_Language::get_flag_informations()` may avoid this inline type?

Reply:
It already has one.

Reply:
And even with that, PHPStan doesn't type `$flags` correctly? Damn ...

Reply:
`PLL_Language::get_flag_informations()` is used to set `$flags['flag']`, it probably has an issue with `$flags['custom_flag']`.

Reply:
I see.